examples_x035/usbpd_sink/usbpd.h

```diff
@@ -0,0 +1,921 @@
+	"\tEPR Mode Capable: %s\n"           \
+	"\tUnchunked Extended Message: %s\n" \
+	"\tDual Role Data: %s\n"             \
+	"\tUSB Communications Capable: %s\n" \
+	"\tUnconstrained Power: %s\n"        \
+	"\tUSB Suspend Supported: %s\n"      \
+	"\tDual Role Power: %s\n"
+
+#define FIXED_SUPPLY_FMT_ARGS( pdo )                                                                  \
+	( ( pdo )->FixedSupply.MaxCurrentIn10mA * 10 ), ( ( pdo )->FixedSupply.VoltageIn50mV * 50 ),      \
+		( ( pdo )->FixedSupply.PeakCurrent ), ( ( pdo )->FixedSupply.EPRModeCapable ? "Yes" : "No" ), \
+		( ( pdo )->FixedSupply.UnchunkedExtendedMessage ? "Yes" : "No" ),                             \
+		( ( pdo )->FixedSupply.DualRoleData ? "Yes" : "No" ),                                         \
+		( ( pdo )->FixedSupply.USBCommunicationsCapable ? "Yes" : "No" ),                             \
+		( ( pdo )->FixedSupply.UnconstrainedPower ? "Yes" : "No" ),                                   \
+		( ( pdo )->FixedSupply.USBSuspendSupported ? "Yes" : "No" ),                                  \
+		( ( pdo )->FixedSupply.DualRolePower ? "Yes" : "No" )
+
+typedef struct
+{
+	uint32_t MaxCurrentIn10mA : 10u;
+	uint32_t MinVoltageIn50mV : 10u;
+	uint32_t MaxVoltageIn50mV : 10u;
+	USBPD_PowerDataObject_e PDOType : 2u; // shall be eUSBPD_PDO_VARIABLE
+} USBPD_VariablePDO_t;
+
+#define VARIABLE_SUPPLY_FMT  \
+	"\nVariable Supply:\n"   \
+	"\tMax Current: %d mA\n" \
+	"\tMin Voltage: %d mV\n" \
+	"\tMax Voltage: %d mV\n"
+
+#define VARIABLE_SUPPLY_FMT_ARGS( pdo )                                                                   \
+	( ( pdo )->VariableSupply.MaxCurrentIn10mA * 10 ), ( ( pdo )->VariableSupply.MinVoltageIn50mV * 50 ), \
+		( ( pdo )->VariableSupply.MaxVoltageIn50mV * 50 )
+
+typedef struct
+{
+	uint32_t MaxPowerIn250mW : 10u;
+	uint32_t MinVoltageIn50mV : 10u;
+	uint32_t MaxVoltageIn50mV : 10u;
+	USBPD_PowerDataObject_e PDOType : 2u; // shall be eUSBPD_PDO_BATTERY
+} USBPD_BatteryPDO_t;
+
+#define BATTERY_SUPPLY_FMT   \
+	"\nBattery Supply:\n"    \
+	"\tMax Power: %d mW\n"   \
+	"\tMin Voltage: %d mV\n" \
+	"\tMax Voltage: %d mV\n"
+
+#define BATTERY_SUPPLY_FMT_ARGS( pdo )                                                                  \
+	( ( pdo )->BatterySupply.MaxPowerIn250mW * 250 ), ( ( pdo )->BatterySupply.MinVoltageIn50mV * 50 ), \
+		( ( pdo )->BatterySupply.MaxVoltageIn50mV * 50 )
+
+typedef struct
+{
+	uint32_t MaxCurrentIn50mA : 7u;
+	uint32_t Reserved_7bit : 1u; // shall be set to zero
+	uint32_t MinVoltageIn100mV : 8u;
+	uint32_t Reserved_16bit : 1u; // shall be set to zero
+	uint32_t MaxVoltageIn100mV : 8u;
+	uint32_t Reserved_25_26bit : 2u; // shall be set to zero
+	uint32_t PPSpowerLimited : 1u;
+	USBPD_AugmentedPDO_e AugmentedType : 2u; // shall be eUSBPD_APDO_SPR_PPS
+	USBPD_PowerDataObject_e PDOType : 2u; // shall be eUSBPD_PDO_AUGMENTED
+} USBPD_SPR_PPS_APDO_t;
+
+#define SPR_PPS_FMT          \
+	"\nPPS Supply:\n"        \
```

Comment:
Same comment, it would be much better for these one-off strings to appear at their usage-site instead of in the core part of the header.